### PR TITLE
Docs: Resolutions documentation minor cleanup

### PR DESCRIPTION
**Pull request by spanishpear**
Was looking around and saw that both of these have been cleaned up.

Replaced with an example resolution, but also happy to remove the file :) 

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

## Greptile Summary

This PR simplifies the RESOLUTIONS.md file by removing specific package version pins and replacing them with an example template entry for documenting package resolutions.

- Removes outdated version pins for `svelte-check` and `overlayscrollbars` from RESOLUTIONS.md
- Adds example template entry to demonstrate proper resolution documentation format
- Maintains file structure while cleaning up obsolete content
- Consider validating removed pins are no longer needed before merging

### CODE REVIEW

File: RESOLUTIONS.md
Comment (SpanishPear) [reply RESOLUTIONS.md:7]: probably fair - will go off whatever the maintainers prefer 🤷

**Review by valentinpalkovic — APPROVED** [review]

**Review by jonniebigodes — APPROVED** [review]
@SpanishPear, thank you for putting together this pull request and helping us improve the documentation by catching this small item. We appreciate it 🙏 !

I've checked, and everything is good on my end as well. 

I'm going to merge this once the checklist clears (cc @valentinpalkovic )

**Pull request merged**
